src/py/flwr/client/app.py

```diff
@@ -22,6 +22,9 @@
 from pathlib import Path
 from typing import Callable, ContextManager, Optional, Tuple, Union
 
+from grpc import RpcError
+from requests.exceptions import ConnectionError as RequestsConnectionError
```

Comment:
We must not depend on `requests` here because most users won't have it installed. Instead of doing the type `if/else` in this module, we could return the error type to check for from the respective connection.

This line

```
 receive, send, create_node, delete_node = conn
```

would change to

```
 receive, send, create_node, delete_node, connection_error_type = conn
```

Reply:
Good catch! I'll do it in the `_init_connection` function